### PR TITLE
fix(sidenav): throw error when sidenav has 2 sidenavs on the same sid…

### PR DESCRIPTION
**Pull request by mmalerba**
…e at the same time

### CODE REVIEW

**Comment by mmalerba** [comment]
fixes #3329

File: src/demo-app/sidenav/sidenav-demo.html [hunk 1]
```diff
@@ -51,10 +51,13 @@
...
 
 <md-sidenav-container class="demo-sidenav-container">
-  <md-sidenav #dynamicAlignSidenav mode="push" [align]="side">Drawer</md-sidenav>
+  <md-sidenav #dynamicAlignSidenav1 mode="side" [align]="invert ? 'end' : 'start'">Start</md-sidenav>
+  <md-sidenav #dynamicAlignSidenav2 mode="side" [align]="invert ? 'start' : 'end'">End</md-sidenav>
 
   <div class="demo-sidenav-content">
-    <button (click)="dynamicAlignSidenav.toggle()">Toggle sidenav</button>
-    <button (click)="side = (side == 'start') ? 'end' : 'start'">Change sides</button>
+    <button (click)="dynamicAlignSidenav1.toggle(); dynamicAlignSidenav2.toggle()">
```
Comment (kara) [inline src/demo-app/sidenav/sidenav-demo.html]: Can we move this into a method?

File: src/lib/sidenav/sidenav.spec.ts [hunk 2]
```diff
@@ -373,23 +373,25 @@ describe('MdSidenav', () => {
...
 
       const testComponent: SidenavDynamicAlign = fixture.debugElement.componentInstance;
-      const sidenavEl = fixture.debugElement.query(By.css('md-sidenav')).nativeElement;
-      expect(sidenavEl.classList).not.toContain('mat-sidenav-invalid');
-
       testComponent.sidenav1Align = 'end';
-      fixture.detectChanges();
 
-      expect(sidenavEl.classList).toContain('mat-sidenav-invalid');
+      expect(() => fixture.detectChanges()).toThrow();
```
Comment (kara) [inline src/lib/sidenav/sidenav.spec.ts]: Check the specific error?

**Comment by mmalerba** [comment]
comments addressed

**Comment by kara** [comment]
@mmalerba  Can you rebase?

**Pull request merged**